packages/angular/cli/src/commands/update/schematic/index.ts

```diff
@@ -311,10 +311,11 @@ function _performUpdate(
     }
   });
 
-  const newContent = JSON.stringify(packageJson, null, 2);
-  if (packageJsonContent.toString() != newContent || migrateOnly) {
+  const newContent =
+    JSON.stringify(packageJson, null, 2) + (packageJsonContent.endsWith('\n') ? '\n' : '');
```

Comment:
The line ending can also be `\r\n` so likely you are looking at something like 
```ts
const eofMatches = packageJsonContent.match(/\r?\n$/);
const eof = eofMatches?.[0];

if (eof) {
  packageJson += eof;
}

newContent = JSON.stringify(packageJson, null, 2);
```

Reply:
Thanks for the review. You're right, I completely forgot about the Windows folks. It is fixed now. Feel free to take a look.